packages/core/src/modules/routing/RecipientModule.ts

```diff
@@ -157,38 +156,54 @@ export class RecipientModule {
         this.logger.warn(
           `Websocket connection to mediator with connectionId '${mediator.connectionId}' is closed, attempting to reconnect...`
         )
-        await this.openMediationWebSocket(mediator)
+        try {
+          await this.openMediationWebSocket(mediator)
+          if (mediator.pickupStrategy === MediatorPickupStrategy.PickUpV2) {
+            // Start Pickup v2 protocol to receive messages received while websocket offline
+            await this.sendStatusRequest({ mediatorId: mediator.id })
```

Comment:
Hmm after thinking about it a bit more this doesn't make fully sense to me. This method is called `initiateImplicitPickup`, while we're actually sending an explicit message to retrieve the pending messages from the queue. The part that is shared is that we open a websocket. Maybe we should rename this method to better reflect the what is happening?